### PR TITLE
Change timeouts to 60000 in measurements and 0 in TestStand fixtures

### PR DESCRIPTION
**Pull request by dixonjoel**
Signed-off-by: Joel Dixon <joel.dixon@ni.com>

- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-python/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Change timeouts for reserving sessions to 60 seconds in the measurement and to 0 in TestStand usage.

### Why should this Pull Request be merged?

#220 
Matches the LabVIEW repo examples.

### What testing has been done?

No manual testing.

### CODE REVIEW

File: examples/nidcpower_source_dc_voltage/measurement.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def measure(
                 context=measurement_service.context.pin_map_context,
                 pin_or_relay_names=pin_names,
                 instrument_type_id=nims.session_management.INSTRUMENT_TYPE_NI_DCPOWER,
-                timeout=-1,
+                timeout=60000,
```
Comment (bkeryan) [inline examples/nidcpower_source_dc_voltage/measurement.py:75]: In Python, timeouts are specified in seconds, so use 60.

Please test one of the examples in both InstrumentStudio and TestStand, including the timeout case.

**Review by bkeryan — CHANGES_REQUESTED** [review]

File: examples/nidcpower_source_dc_voltage/measurement.py (hunk 1 above)
Comment (bkeryan) [reply examples/nidcpower_source_dc_voltage/measurement.py:75]: FYI: the type annotation says `Optional[float]` and `60` is an `int`, but that's ok: https://mypy.readthedocs.io/en/latest/duck_type_compatibility.html

File: examples/nivisa_dmm_measurement/teststand_fixture.py [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
 
-from _helpers import GrpcChannelPoolHelper, PinMapClient
-from _visa_helpers import (
-    INSTRUMENT_TYPE_DMM_SIMULATOR,
-    create_visa_resource_manager,
-    create_visa_session,
-    log_instrument_id,
-    reset_instrument,
-)
+from _helpers import GrpcChannelPoolHelper
```
Comment (pbirkhol-ni) [inline examples/nivisa_dmm_measurement/teststand_fixture.py:3]: Can you revert this change? [PEP8 ](https://peps.python.org/pep-0008/#imports)has no problem with importing multiple objects from the same file. I think this will also fix the PR build.

**Review by pbirkhol-ni — APPROVED** [review]

File: examples/nivisa_dmm_measurement/teststand_fixture.py (hunk 2 above)
Comment (pbirkhol-ni) [reply examples/nivisa_dmm_measurement/teststand_fixture.py:3]: I meant to highlight lines 3-9.

Comment (mshafer-NI) [reply examples/nivisa_dmm_measurement/teststand_fixture.py:3]: @dixonjoel, you may need to update the version of ni-python-styleguide in the lock file / your venv (0.4.0 keeps these together now)

**Comment by bkeryan** [comment]
Let's also add comments explaining the timeouts. Something like this:
- teststand_fixture.py: This code module sets up the sessions, so error immediately if they are in use.
- measurement.py: If another measurement is using the session, wait for it to complete. Specify a timeout to aid in debugging missed unreserve calls. Long measurements may require a longer timeout.

File: examples/nivisa_dmm_measurement/teststand_fixture.py (hunk 2 above)
Comment (dixonjoel) [reply examples/nivisa_dmm_measurement/teststand_fixture.py]: I updated the pyproject.toml to have >=0.4.0 for ni-python-styleguide. This required me to run poetry lock, so please check the lock file and pyproject file.

I have Poetry (version 1.3.1). I can't remember the correct version, but I remember @mshafer-NI saying I had to have a certain version to get the .lock file right.

Comment (dixonjoel) [reply examples/nivisa_dmm_measurement/teststand_fixture.py]: Also, I'm getting a bunch of D417 errors for docstrings. I'm not sure why since the methods mostly have documentation on them. Unless I need to document the 'self' argument or the docs are formatted incorrectly.
> ./examples/niswitch_control_relays/teststand_fixture.py:[10](https://github.com/ni/measurementlink-python/actions/runs/4098613934/jobs/7067851096#step:9:11):1: D417 Missing argument descriptions in the docstring
./examples/niswitch_control_relays/_helpers.py:52:1: D417 Missing argument descriptions in the docstring
./examples/niswitch_control_relays/_helpers.py:100:1: D417 Missing argument descriptions in the docstring
./examples/nivisa_dmm_measurement/teststand_fixture.py:19:1: D417 Missing argument descriptions in the docstring

File: examples/nidcpower_source_dc_voltage/measurement.py (hunk 1 above)
Comment (dixonjoel) [reply examples/nidcpower_source_dc_voltage/measurement.py]: Any good ideas about how to test the timeout case? My thought was to build in a 60 second delay in the session manager service in my local build before it checks the timeout so it would always time out.

I figured if I just killed or made the session manager service missing, it would error out before it had a chance to timeout during the reserve session call.

File: examples/nivisa_dmm_measurement/teststand_fixture.py (hunk 2 above)
Comment (mshafer-NI) [reply examples/nivisa_dmm_measurement/teststand_fixture.py]: maybe related?
https://github.com/PyCQA/pydocstyle/issues/443

> I have Poetry (version 1.3.1). I can't remember the correct version, but I remember @mshafer-NI saying I had to have a certain version to get the .lock file right.

@bkeryan this project is using Poety 1.2, right?

File: examples/nidcpower_source_dc_voltage/measurement.py (hunk 1 above)
Comment (bkeryan) [reply examples/nidcpower_source_dc_voltage/measurement.py]: Run another measurement service that uses DCPower. Add a delay to that service or attach a debugger to it.

File: examples/nivisa_dmm_measurement/teststand_fixture.py (hunk 2 above)
Comment (bkeryan) [reply examples/nivisa_dmm_measurement/teststand_fixture.py]: I think the D417 errors occur because we're mixing numpy and Google style.

numpy uses dashes with no colon: https://numpydoc.readthedocs.io/en/latest/format.html
```
Parameters
----------
x : type
    Description of parameter `x`.
y
    Description of parameter `y` (with type not specified).
```

Google style uses a colon with no dashes: https://google.github.io/styleguide/pyguide.html
```
    """Connects to the next available port.

    Args:
      minimum: A port value greater or equal to 1024.

    Returns:
      The new minimum port.

    Raises:
      ConnectionError: If no available port is found.
    """
```

Comment (bkeryan) [reply examples/nivisa_dmm_measurement/teststand_fixture.py]: I think you should split the ni-python-styleguide update into a separate PR.

Comment (dixonjoel) [reply examples/nivisa_dmm_measurement/teststand_fixture.py]: I'm fine with that. That just means these lines will have to be separate imports until we migrate the style guide.

Comment (bkeryan) [reply examples/nivisa_dmm_measurement/teststand_fixture.py]: You don't have to revert the imports. `ni-python-styleguide lint` never required these imports to be on separate lines.

File: examples/nidcpower_source_dc_voltage/measurement.py (hunk 1 above)
Comment (dixonjoel) [reply examples/nidcpower_source_dc_voltage/measurement.py]: Tested with delay in InstrumentStudio and TestStand. InstrumentStudio - errors out after the specified delay (60 seconds). TestStand - errors out immediately.

**Review by bkeryan — APPROVED** [review]
